### PR TITLE
Namoray/refactor

### PR DESCRIPTION
**Pull request by wanderingweights**

### CODE REVIEW

**Comment by namoray** [comment]
Lord help us. Sorry not sorry.

Changes:
- Changed 'Task' to 'RawTask'. Not happy with the name though. 
- RawTask tidied up a lot of the columns. Made all the axolotl columns compatible with so
- Added account_id to all the endpoints, and tables
- Renamed the columns in the db
- Changed get_all_tasks to instead get all tasks by a specific account id (do we need the other endpoint)
- Refactoring of various endpoints for various reasons
- pre-commit run on all (is someone still not running it?)
- Improved setup for valis by making bootstrap a `task`

File: validator/endpoints/tasks.py [hunk 1]
```diff
@@ -218,76 +190,22 @@ async def get_leaderboard(
...
-    )
-
-    router.add_api_route(
-        "/v1/tasks/node_results/{hotkey}",
-        get_node_results,
-        response_model=AllOfNodeResults,
-        tags=["Training"],
-        methods=["GET"],
-    )
+    router.add_api_route(TASKS_CREATE_ENDPOINT, create_task, methods=["POST"])
```
Comment (wanderingweights) [inline validator/endpoints/tasks.py:203]: Did wonder why we even have response_models when it's already explicit within the task

File: validator/tasks/scheduler.py [hunk 2]
```diff
@@ -24,59 +22,69 @@
...
-    response: list[dict[str, Any]] = await call_content_service(csts.GET_RANDOM_MODELS_ENDPOINT, keypair)
-    models = [ModelData(**model) for model in response]
-    random.shuffle(models)
-    for m in models:
-        yield m.model_id
+    response = await call_content_service(cst.GET_RANDOM_MODELS_ENDPOINT, keypair)
+    if not isinstance(response, list):
+        raise TypeError("Expected a list of responses from GET_ALL_MODELS_ENDPOINT")
+    models: list[dict[str, Any]] = response
+    TEMP_MODEL_FAMILIES_ACCEPTED = ["qwen", "llama", "falcon", "mistral", "gemma", "gemini", "phi"]
```
Comment (wanderingweights) [inline validator/tasks/scheduler.py:29]: These bits were removed - no longer needed 

async def _get_models(keypair: Keypair) -> AsyncGenerator[str, None]:
    response: list[dict[str, Any]] = await call_content_service(csts.GET_RANDOM_MODELS_ENDPOINT, keypair)
    models = [ModelData(**model) for model in response]
    random.shuffle(models)
    for m in models:
        yield m.model_id

File: validator/tasks/scheduler.py [hunk 3]
```diff
@@ -24,59 +22,69 @@
...
-    random.shuffle(datasets)
-    for ds in datasets:
-        yield ds.dataset_id
-
-
-async def _get_the_columns_for_dataset(dataset_id: str, keypair: Keypair) -> DatasetRequest:
-    url = csts.GET_COLUMNS_FOR_DATASET_ENDPOINT.replace(
-        "{dataset}", dataset_id)
+    response = await call_content_service(cst.GET_RANDOM_DATASETS_ENDPOINT, keypair)
+    if not isinstance(response, list):
```
Comment (wanderingweights) [inline validator/tasks/scheduler.py:42]: AsyncGenerator[str, None]:
    response: list[dict[str, Any]] = await call_content_service(csts.GET_RANDOM_DATASETS_ENDPOINT, keypair)
    datasets = [DatasetData(**ds) for ds in response]
    random.shuffle(datasets)
    for ds in datasets:
        yield ds.dataset_id
        
        Cleaner?

File: validator/endpoints/tasks.py [hunk 4]
```diff
@@ -31,122 +30,62 @@
 logger = get_logger(__name__)
 
 
+TASKS_CREATE_ENDPOINT = "/v1/tasks/create"
+GET_TASKS_BY_ACCOUNT_ENDPOINT = "/v1/tasks/account/{account_id}"
```
Comment (wanderingweights) [inline validator/endpoints/tasks.py:34]: don't shit on your beaut of a refactor, but feels weird these are here and not in a constants somewhere.. just me

**Pull request closed without merging**